src/v/datalake/coordinator/state.h

```diff
@@ -81,10 +81,31 @@ struct partition_state
 struct topic_state
   : public serde::
       envelope<topic_state, serde::version<0>, serde::compat_version<0>> {
-    auto serde_fields() { return std::tie(pid_to_pending_files); }
-
+    auto serde_fields() {
+        return std::tie(revision, pid_to_pending_files, lifecycle_state);
+    }
+
+    enum class lifecycle_state_t {
+        // ready to accept new files
+        live,
+        // topic deleted, new files can't be accepted (but already accepted
+        // files will be committed)
```

Comment:
I'm wondering if this is a hard requirement, and whether we can relax it in favor of a simple state machine where live is the only state where pending entries are allowed. I.e. it doesn't seem unreasonable to just drop uncommitted files, and enforce that closed has no pending entries.

From the perspective of an admin, I don't think I'd really expect any additional commits after dropping the topic (regardless of purging)

Reply:
Sure, we can move them to orphaned

Reply:
After a discussion we decided to remove the orphaned collection from the STM and commit the already added files.